addons/mass_mailing/models/mailing.py

```diff
@@ -650,3 +650,26 @@ def _image_to_url(b64image: bytes):
             return lxml.html.tostring(root)
 
         return body_html
+
+    def _prepare_mail_body(self, body=None):
```

Comment:
What about moving it to the mail.render.mixin so that it is available as a tool ? We could name it _prepend_preview, working as api.model with body as a real parameter.